src/server.py

```diff
@@ -605,5 +588,105 @@ def template(filename):
         return response_body
     abort(404)
 
+
+def update_preseed():
+    """ we sync preseed.json to jobpririties in server on startup, since that is
+        the only time we expect preseed.json to change. """
+
+    # get preseed data first
+    preseed_path = os.path.join(os.path.dirname(SCRIPT_DIR), 'src', 'preseed.json')
+    preseed = []
+    with open(preseed_path, 'r') as fHandle:
+        preseed = json.load(fHandle)
+
+    # Preseed data will have fields: buildtype,testtype,platform,priority,timeout,expires
+    # The expires field defaults to 2 weeks on a new job in the database
+    # Expires field has a date "YYYY-MM-DD", but can have "*" to indicate never
+    # Typical priority will be 1, but if we want to force coalescing we can do that
+    # One hack is that if we have a * in a buildtype,testtype,platform field, then
+    # we assume it is for all flavors of the * field: i.e. linux64,pgo,* - all tests
+    # assumption - preseed fields are sanitized already - move parse_testtype to utils.py ?
+    for job in preseed:
+        _buildsystem = "buildbot" # use this as a default
```

Comment:
could you tell me the usage of ** * ** ? And what will happened when the build system is "taskcluster".